code/modules/mob/animations.dm

```diff
@@ -241,3 +240,17 @@ note dizziness decrements automatically in the mob's Life() proc.
 			set_dir(D)
 			spintime -= speed
 	return
+
+/mob/living/proc/on_structure_offset(var/offset = 0)
+	if(offset)
+		var/check = default_pixel_z + offset
+		if(pixel_z != check)
+			animate(src, pixel_z = check, time = 2, easing = SINE_EASING)
+	else
+		var/obj/structure/S = (locate() in get_turf(src))
+		if(!S && pixel_z != default_pixel_z)
+			animate(src, pixel_z = default_pixel_z, time = 2, easing = SINE_EASING)
+
+/mob/living/Move()
+	. = ..()
+	on_structure_offset(0)
```

Comment:
As I mentioned on discord:

I do not like this.
This loops through the contents of every turf the mob moves into to check if there is a table.
Cant the reset be done with [Uncrossed?](http://www.byond.com/docs/ref/info.html#/atom/proc/Uncrossed)